mudlet-mapper.xml

```diff
@@ -9808,49 +9788,102 @@ end</script>
 </script>
 					<eventHandlerList />
 					<Script isActive="yes" isFolder="no">
-						<name>register_midkemias_envdata</name>
+						<name>register_starmourn_envdata</name>
 						<packageName></packageName>
-						<script>function register_midkemias_envdata(_, game)
-  if game ~= "Midkemia Online" then return end
+						<script>function register_starmourn_envdata(_, game)
+  if game ~= "Starmourn" then
+	print("bye")
+    return
+  end
+	print("hi")
   mmp.envids =
-  {
-    ["constructed underground"] = 2,
-    ["fresh water"]             = 20,
-    ["natural underground"]     = 3,
-    ["shores of Lake Moraelin"] = 33,
-    beach                       = 5,
-    canyon                      = 29,
-    farmlands                   = 31,
-    forest                      = 4,
-    garden                      = 19,
-    grasslands                  = 7,
-    hills                       = 9,
-    ice                         = 27,
-    mines                       = 28,
-    mountains                   = 14,
-    ocean                       = 18,
-    path                        = 11,
-    plains                      = 30,
-    reef                        = 23,
-    river                       = 10,
-    road                        = 12,
-    ruins                       = 24,
-    sewers                      = 21,
-    swamp                       = 15,
-    temple                      = 26,
-    tundra                      = 16,
-    urban                       = 8,
-    valley                      = 13,
-    village                     = 25,
-    volcanic                    = 32,
-  }
-
-  mmp.waterenvs = {}
-  local waterids = { "river", "ocean", "fresh water", "reef" }
-  for i = 1, #waterids do mmp.waterenvs[mmp.envids[waterids[i]]] = true end
-
-  mmp.envidsr = {};
-  for name, id in pairs(mmp.envids) do mmp.envidsr[id] = name end
+    {
+      ["mountain"] = 173,
+      ["street"] = 116,
+      ["mine"] = 209,
+      ["urban"] = 116,
+      ["building"] = 116,
```

Comment:
These double environment IDs are not a good idea. IIRC it leads to the environments not being correctly loaded from a binary map. See #16

Reply:
https://www.starmourn.com/maps/map.xml

Reply:
You used colours, not environments ;)

Reply:
Yeah, environments are colours...

Reply:
Colours of the rooms. See https://www.achaea.com/maps/map.xml

Reply:
You want the environment ID on the very left, not the colour.

Reply:
Why would I want that haha?

I think we have some confusion here - have you tried the script and verified that the environments don't show up correctly?

Reply:
Yes, have tried it and it doesn't work properly. You want the IDs, because the IDs are what defines environment in the map, not the colours. For instance, if you ctrl+f the rooms in the map, you won't find an environment 116, because that's the environment colour, not the ID used to distinguish environment. Is the same way with Achaea's map.

Though, when parsing the map download, mudlet seems to read all colours as black. Probably because Starmourn's map is using xterm256 colours instead of the base 16.

Reply:
I stand corrected, you're right